tcf_website/views/search.py

```diff
@@ -17,14 +18,61 @@ def search(request):
     courses = fetch_courses(query)
     instructors = fetch_instructors(query)
 
+    courses_first = decide_order(query, courses, instructors)
+
     # Set arguments for template view
-    args = set_arguments(query, courses, instructors)
+    args = set_arguments(query, courses, instructors, courses_first)
     context_vars = args
 
     # Load template view
     return render(request, 'search/search.html', context_vars)
 
 
+def decide_order(query, courses, instructors):
+    """Decides if courses or instructors should be displayed first."""
+    # Returns True if courses should be prioritized, False if instructors should be prioritized
+
+    # Calculates z-score of top courses result
+    if len(courses['results']) > 1:
+        course_scores = [x['score'] for x in courses['results']]
+
+        courses_mean = statistics.mean(course_scores)
+
+        courses_stddev = statistics.stdev(course_scores, courses_mean)
+
+        if courses_stddev == 0:
+            courses_stddev = 1
+        courses_z = courses['results'][0].get('score') - courses_mean
```

Comment:
Are the search results guaranteed to be sorted descending by score? Intuitively it makes sense, just wondering if that's confirmed anywhere.

Reply:
Yep, it is sorted from top to bottom. I printed out scores when testing to find that out.